src/shell/Commands/ExtensionCommand.Unregister.cs

```diff
@@ -0,0 +1,59 @@
+using McMaster.Extensions.CommandLineUtils;
+using System.IO.Abstractions;
+using Newtonsoft.Json.Linq;
+
+namespace NeoShell.Commands
+{
+    partial class ExtensionCommand
+    {
+        [Command("unregister", Description = "Unregisters an extension")]
+        internal class Unregister
+        {
+            private const string BaseFolder = ".neo";
+            private const string ExtensionsFile = "extensions.json";
+            private readonly IFileSystem fileSystem;
+
+            public Unregister(IFileSystem fileSystem)
+            {
+                this.fileSystem = fileSystem;
+            }
+
+            [Argument(0, Description = "Display of the extension")]
+            internal string Name { get; init; } = string.Empty;
+
+
+            internal int OnExecute(CommandLineApplication app, IConsole console)
+            {
+                try
+                {
+                    string rootPath = this.fileSystem.Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), BaseFolder);
+
+                    string extensionsFilePath = this.fileSystem.Path.Combine(rootPath, ExtensionsFile);
+                    string extensionsJsonContent = this.fileSystem.File.ReadAllText(extensionsFilePath);
+                    JArray extensionsArray = JArray.Parse(extensionsJsonContent);
```

Comment:
Should we be mapping the loosely typed JSON to C# types? Not sure it's worth it, but figured we should discuss

Reply:
Refactored the code to use existing ShellExtensions and ShellExtenion classes. Thank you for reminding me those exist.